### ISSUE TITLE
Use new /migrated endpoint in the meditor

### ISSUE THREAD

**Issue by dchiquito** [dandi/dandi-archive#737]
See https://github.com/dandi/dandi-api/blob/master/doc/design/draft-metadata-migration.md

The meditor should use the new `GET api/dandisets/{dandiset_pk}/versions/{version}/migrated/` endpoint to ensure that saving the meditor always migrates the metadata.

Currently blocked on https://github.com/dandi/dandi-api/issues/565 to add the endpoint

**Comment by AlmightyYakob** [comment]
Closing in favor of https://github.com/dandi/dandi-archive/issues/565
